### PR TITLE
Feature/gh 10117 cypress test bundle types

### LINKED ISSUE #10117: Enhance Cypress test with new ui types
Enhance existing update configuration CPQ cypress test with new ui types.

### PR DESCRIPTION
**Pull request by Uli-Tiger**
[GH-10117]

- adds missing Id's in some UI Components
- renamed (shorten) almost all constants in cypress test, so that most statements are formatted in one line. Value and Attribute names are provided via jsdoc comment, so one can see whats its meant on mouse over.  
- enhanced configuration page methods to be able to handle new UI components when selecting values and or getting selection state.
- added 3 new tests for the new UI components

### CODE REVIEW

File: feature-libs/product-configurator/rulebased/components/attribute/types/multi-selection-bundle/configurator-attribute-multi-selection-bundle.component.html [hunk 1]
```diff
@@ -1,5 +1,9 @@
-<ng-container *ngIf="attribute?.values?.length">
```
Comment (Larisa-Staroverova) [inline feature-libs/product-configurator/rulebased/components/attribute/types/multi-selection-bundle/configurator-attribute-multi-selection-bundle.component.html:1]: I would the **ng-container** element not remove. This element is not visibale in HTML tree, it only verifies the if-condition. If the statement is true, only then the inner HTML elements will be created/ added to the HTML tree. By means of this statement the HTML tree stays lean

File: feature-libs/product-configurator/rulebased/components/attribute/types/single-selection-bundle/configurator-attribute-single-selection-bundle.component.html [hunk 2]
```diff
@@ -1,4 +1,7 @@
-<ng-container *ngIf="attribute?.values?.length">
```
Comment (Larisa-Staroverova) [inline feature-libs/product-configurator/rulebased/components/attribute/types/single-selection-bundle/configurator-attribute-single-selection-bundle.component.html:1]: The same hee. (see the previous comment)

File: projects/storefrontapp-e2e-cypress/cypress/integration/regression/cpq-configuration.e2e-spec.ts [hunk 3]
```diff
@@ -1,37 +1,63 @@
...
-const ACCESSORIES = 'Accessories';
-const INSURANCE_AND_WARRANTY = 'Insurance and Warranty';
-
-// List of attributes
-const attributeHeaders = {
-  mainComponents: ['Camera Body', 'Memory Card', 'Lenses'],
-  accessories: ['Tripod', 'Bag', 'Special Accessories'],
-  insuranceAndWarranty: [
+const RB = 'radioGroup';
+const CBL = 'checkBoxList';
```
Comment (Larisa-Staroverova) [inline projects/storefrontapp-e2e-cypress/cypress/integration/regression/cpq-configuration.e2e-spec.ts]: According to my point of view, the constants names are **too short**. 
We cann't expect from everyone that he or she knows CPQ product by heart.

File: projects/storefrontapp-e2e-cypress/cypress/integration/regression/cpq-configuration.e2e-spec.ts [hunk 4]
```diff
@@ -53,149 +79,150 @@ context('CPQ Configuration', () => {
...
-      configuration.goToPDPage(powertoolsShop, testProduct);
+      configuration.goToPDPage(POWERTOOLS, PROD_CODE_CAM);
       configuration.clickOnConfigureBtnInCatalog();
     });
   });
 
-  describe('Update attribute values', () => {
-    it('should support update values for radio button attribute type', () => {
-      configuration.goToPDPage(powertoolsShop, testProductCoffeeMachine);
+  describe('UiTypes', () => {
```
Comment (Larisa-Staroverova) [inline projects/storefrontapp-e2e-cypress/cypress/integration/regression/cpq-configuration.e2e-spec.ts]: Could you write longer description? For instance: **Handling different UI types**

**Review by Larisa-Staroverova — CHANGES_REQUESTED** [review]
Could you have a look at my comments?

File: feature-libs/product-configurator/rulebased/components/attribute/types/multi-selection-bundle/configurator-attribute-multi-selection-bundle.component.html (hunk 1 above)
Comment (Uli-Tiger) [reply feature-libs/product-configurator/rulebased/components/attribute/types/multi-selection-bundle/configurator-attribute-multi-selection-bundle.component.html:1]: an ng-container can't have an id an div can have an id, that's why i replaced the container with an div. I assume that the div is not rendered, as well, if the if is false?

File: projects/storefrontapp-e2e-cypress/cypress/integration/regression/cpq-configuration.e2e-spec.ts (hunk 3 above)
Comment (Uli-Tiger) [reply projects/storefrontapp-e2e-cypress/cypress/integration/regression/cpq-configuration.e2e-spec.ts]: These 2 terms are not related to CPQ? RB = RadioButton. Ok maybe RG= radioGroup would be better?
What would be a suitable contestant name in your eyes?

Comment (Larisa-Staroverova) [reply projects/storefrontapp-e2e-cypress/cypress/integration/regression/cpq-configuration.e2e-spec.ts]: Maybe, the names should reflect the UI names in a certain way in order to find them better in the UI

**Review by Larisa-Staroverova — APPROVED** [review]

**Pull request merged**